subprojects/gradle/instrumentation-tests/src/integTest/kotlin/com/avito/instrumentation/executing/TestExecutorIntegrationTest.kt

```diff
@@ -0,0 +1,99 @@
+package com.avito.instrumentation.executing
+
+import com.avito.instrumentation.configuration.target.TargetConfiguration
+import com.avito.instrumentation.configuration.target.scheduling.quota.QuotaConfiguration
+import com.avito.instrumentation.createStubInstance
+import com.avito.instrumentation.report.listener.FakeTestReporter
+import com.avito.instrumentation.reservation.client.kubernetes.KubernetesReservationClient
+import com.avito.instrumentation.reservation.client.kubernetes.createStubInstance
+import com.avito.instrumentation.reservation.devices.provider.KubernetesDevicesProvider
+import com.avito.instrumentation.reservation.request.Device
+import com.avito.instrumentation.reservation.request.Reservation
+import com.avito.instrumentation.reservation.request.createStubInstance
+import com.avito.instrumentation.suite.model.TestWithTarget
+import com.avito.report.model.TestStaticDataPackage
+import com.avito.report.model.createStubInstance
+import com.avito.runner.service.worker.device.adb.Adb
+import com.avito.runner.service.worker.device.adb.AdbDevicesManager
+import com.avito.utils.logging.CILogger
+import com.avito.utils.logging.FakeCILogger
+import com.avito.utils.logging.commonLogger
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Timeout
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+import java.util.concurrent.TimeUnit.SECONDS
+
+internal class TestExecutorIntegrationTest {
+
+    private val logger = FakeCILogger()
+
+    @Timeout(10, unit = SECONDS)
```

Comment:
We can define also a default timeout for all tests as a fallback. The same problem can occur in other tests.

`junit.jupiter.execution.timeout.test.method.default`
https://junit.org/junit5/docs/5.5.0-RC2/api/org/junit/jupiter/api/Timeout.html

Reply:
i've added 5 min global timeout for all tests and lifecycle methods via system property;
tested it with 1 sec timeout, test fails with InterruptedException like on screenshot
maximum test duration on last executions was 4min 11sec

![image](https://user-images.githubusercontent.com/1105133/102364570-a7baa180-3fc7-11eb-9793-1371279b8db0.png)

Reply:
If we don't have any other `junit.jupiter.execution.*` properties, I have no idea.

Reply:
![image](https://user-images.githubusercontent.com/1105133/102365496-ac338a00-3fc8-11eb-8d65-1754f4a7f67e.png)

just now, in https://github.com/avito-tech/avito-android/pull/677/commits/11727a5182bcf579221e670d3a2383ff763ee707
sorry for confusion